ontopy/excelparser.py

```diff
@@ -0,0 +1,267 @@
+                    warnings.warn(
+                        f"Error in Property assignment for: {concept}. "
+                        f"Property to be Evaluated: {prop}. "
+                        f"Error is {err}."
+                    )
+
+    onto, catalog = get_metadata_from_dataframe(metadata, onto)
+    # Synchronise Python attributes to ontology
+    onto.sync_attributes(
+        name_policy="uuid", name_prefix="EMMO_", class_docstring="elucidation"
+    )
+    onto.dir_label = False
+
+    return onto, catalog
+
+
+# To test: with and without ontology as input
+def get_metadata_from_dataframe(  # pylint: disable=too-many-locals,too-many-branches,too-many-statements
+    metadata: pd.DataFrame,
+    onto: owlready2.Ontology = None,
+    base_iri_from_metadata: bool = True,
+    catalog: dict = None,
+) -> Tuple[owlready2.Ontology, dict]:
+    """
+    Populate ontology with metada from pd.DataFrame
+    """
+
+    if onto is None:
+        onto = get_ontology()
+
+    # base_iri from metadata if it exists and base_iri_from_metadata
+    if base_iri_from_metadata:
+        try:
+            base_iris = _parse_metadata_string(metadata, "Ontology IRI")
+            if len(base_iris) > 1:
+                warnings.warn(
+                    "More than one Ontology IRI given. " "The first was chosen."
+                )
+            base_iri = base_iris[0] + "#"
+            onto.base_iri = base_iri
+        except (TypeError, ValueError, AttributeError):
+            pass
+
+    # Get imported ontologies from metadata
+    try:
+        imported_ontology_paths = _parse_metadata_string(
+            metadata, "Imported ontologies"
+        )
+    except (TypeError, ValueError, AttributeError):
+        imported_ontology_paths = []
+    # Add imported ontologies
+    catalog = {} if catalog is None else catalog
+    for path in imported_ontology_paths:
+        imported = onto.world.get_ontology(path).load()
+        onto.imported_ontologies.append(imported)
+        catalog[imported.base_iri.rstrip("/")] = path
+
+    # Add title
+    try:
+        titles = _parse_metadata_string(metadata, "Title")
+        if len(titles) > 1:
+            warnings.warn(
+                "More than one title is given. " "The first was chosen."
+            )
+        onto.metadata.title.append(english(titles[0]))
+    except (TypeError, ValueError, AttributeError):
+        pass
+
+    # Add versionINFO
+    try:
+        version_infos = _parse_metadata_string(
+            metadata, "Ontology version Info"
+        )
+        if len(titles) > 1:
+            warnings.warn(
+                "More than one versionINFO is given. " "The first was chosen."
+            )
+        onto.metadata.versionInfo.append(english(version_infos[0]))
+    except (TypeError, ValueError, AttributeError):
+        pass
+
+    # Add versionINFO
+    try:
+        licenses = _parse_metadata_string(metadata, "License")
+        if len(licenses) > 1:
```

Comment:
If several licenses are given in the excel sheet, I think it is fair to add them all to the ontology.

It is technically possible to add multiple licenses to an ontology.